azurerm/internal/services/compute/parse/dedicated_host_test.go

```diff
@@ -0,0 +1,90 @@
+package parse
+
+import (
+	"testing"
+)
+
+func TestDedicatedHostID(t *testing.T) {
+	testData := []struct {
+		Name          string
+		Input         string
+		ExpectedOK    bool
+		ExpectedValue *DedicatedHostId
```

Comment:
Could we make this like the supplied PR?
```
Name     string
		Input    string
		Expected *AppServiceResourceID
```

we can assume nil means an error occured

Reply:
The way did in PR,  by looking through the test table, it feels like user of this function can just reply on the first return value to see whether something goes wrong, and there is no guarantee about the second return value (`err`).

While actually we should explicitly tell user some error occurs for invalid resource ID so that user could reply on the `err` to see whether something goes wrong and can use the `err` to log or whatever.